Modules/rat_rescue.py

```diff
@@ -862,35 +880,30 @@ def unmark_delete(self) -> None:
         self.marked_for_deletion.reason = None
         self.marked_for_deletion.reporter = None
 
-    @contextmanager
-    def change(self):
+    @asynccontextmanager
+    async def change(self):
         """
         Convenience method for making safe attribute changes.
 
-        FIXME: currently just ensures rescue.updated_at is updated.
-
-        TODO: replace with Board context manager once its implemented
-
-        TODO: replace current context manager with a dummy once the Board
-            context manager is a thing.
-
-        TODO: implement API integration (probably in the board Contextmanager
-
-        Returns:
-            contextManager
+        Yields:
+            RatBoard: RatBoard this rescue is assigned to
 
 
         Examples:
             ```
 
-            with rescue.change():
+            async with rescue.change():
                 rescue.client = foo
 
             ```
         """
-        yield
-        self.updated_at = datetime.utcnow()
+        log.debug(f"entering change context manager for rescue #{self.api_id}({self.client})")
+
+        yield self.rat_board
+        if len(self.modified_attrs) != 0:
+            self.updated_at = datetime.utcnow()
+        else:
+            log.warning(f"rescue #{self.api_id}({self.client}) was not changed within .change!")
 
-    # TODO: to/from json
-    # TODO: track changes
-    # TODO: helper method for adding / editing quotes
+        if self.rat_board is not None:
+            await self.rat_board.update(rescue=self)
```

Comment:
I have a bit of a headache about this method and the associated `RatBoard.update`.

Why does the board have to be involved in updating the rescue? I don't believe any data from `RatBoard` enters into it. It seems to me that this needlessly complicates the process. Apart from making things more straightforward, leaving the updating in `Rescue.change` would: 
* Allow the updating of rescues not attached to a board (closed ones) which is probably a needed feature.
* Less importantly, it would avoid making `modified_attrs` public, which I don't believe it needs to be.

In the same spirit, I also believe that yielding the board is confusing and not very useful at all.

Reply:
Methinks i did it this way as i originally intended the board to be the thing interfacing with the API rather than the `Rescue` objects. That said, upon consideration perhaps it is indeed a bit of a over-complication.

The question becomes how do we intend the flow of information to work?
- If the rescue is not on a board, then wouldn't it not be possible to update the rescue / method wouldn't do anything?

 - If we make Rescue directly update the API, the Rescue object will need a reference to the API. im thinking this could be accomplished by adding it to the `Rescue` constructor, with a default of `None`. Rescues would be created through their respective boards (which would include closed ones?) which would fill in the API reference.

Reply:
I think imma put this Pr on hold until after the Event system is completed, then i can remove the headache when it comes to who does what in terms of updating a rescue (will result in the board loosing its `update` method / retooling it for something else.